packages/headless/src/features/recent-queries/recent-queries-slice.ts

```diff
@@ -0,0 +1,31 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {
+  registerRecentQueries,
+  clearRecentQueries,
+} from './recent-queries-actions';
+import {executeSearch} from '../search/search-actions';
+import {getRecentQueriesInitialState} from './recent-queries-state';
+
+export const recentQueriesReducer = createReducer(
+  getRecentQueriesInitialState(),
+  (builder) => {
+    builder
+      .addCase(registerRecentQueries, (state, action) => {
+        state.queries = action.payload.queries;
+        state.maxQueries = action.payload.maxQueries;
+      })
+      .addCase(clearRecentQueries, (state) => {
+        state.queries.length = 0;
+      })
+      .addCase(executeSearch.fulfilled, (state, action) => {
+        const query = action.payload.queryExecuted;
```

Comment:
Should add a check here for empty query.

Also, should there be any mechanism for extremely large query ? What if someones inputs a huge string here, do we want to ellipsis the content ?

Reply:
On the UI side I think adding an ellipsis is likely to happen but how would that work here? We need to save the exact query string to be able to re-execute it right?

Reply:
You are right !